serveradmin/serverdb/views.py

```diff
@@ -87,38 +100,35 @@ def history(request):
     if not object_id:
         raise Http404
 
-    adds = ChangeAdd.objects.filter(server_id=object_id)
-    updates = ChangeUpdate.objects.filter(server_id=object_id)
-    deletes = ChangeDelete.objects.filter(server_id=object_id).select_related()
+    # TODO: Change data model so that server_id is moved to ChangeCommit.
+    #       To me this would make most sense. If that is not possible (why?!)
+    #       write a proper (custom) query that performs.
+    where = {'server_id': object_id}
+    adds = ChangeAdd.objects.filter(**where)
+    updates = ChangeUpdate.objects.filter(**where)
+    deletes = ChangeDelete.objects.filter(**where)
 
     # @TODO Transform to PostgreSQL JSON and put a index here
     if search_string:
         adds = adds.filter(attributes_json__contains=search_string)
         updates = updates.filter(updates_json__contains=search_string)
         deletes = deletes.filter(attributes_json__contains=search_string)
 
-    adds = adds.select_related()
-    updates = updates.select_related()
-
     if commit_id:
         adds = adds.filter(commit__pk=commit_id)
         updates = updates.filter(commit__pk=commit_id)
         deletes = deletes.filter(commit__pk=commit_id)
 
+    related = ('commit__app', 'commit__user')
+    adds = adds.select_related(*related)
```

Comment:
On another note, did you also try to use `prefetch_related` instead of `select_related`? It is using more queries, however one can also select `many_to_one` and `many_to_many` relations with that. Maybe interesting to know for future improvements on other columns.

Reply:
you are generally right. but this is a one-to-one relation and we can save an additional query here compared to `prefetch_related`. i still highly recommend `prefetch_related` for all many relations (see `serveradmin/serverdb/query_materializer.py`)

Reply:
I understand, that for one-to-one relations one should use `select_related`, I was just curious how much slower `prefetch_related` would be compared to `select_related`. ;) Was asking if just by any chance you tested that. If not, then not. ^^

Reply:
yeah i tested it in a way that i saw an extra unnecessary query. i haven't measured it though. it probably finishes very fast in any case